src/nerc_rates/models.py

```diff
@@ -53,6 +62,22 @@ def validate_no_overlap(cls, data: Self):
 
         return data
 
+    @pydantic.model_validator(mode="after")
+    @classmethod
+    def validate_rate_type(cls, data: Self):
+        rate_type = {"str": str, "bool": bool, "Decimal": Decimal}.get(
```

Comment:
This should technically be:

```suggestion
        rate_type = {RateType.STR: str, RateType.BOOL: bool, RateType.DECIMAL: Decimal}.get(
```

We use enums to avoid hardcoding strings. If there were a typo in one of your strings, it would be hard to detect. On the other hand, if you mistyped something like `RateType.BOL`, that error would be detected much earlier.

But we can change that in a future PR.

Reply:
@KelvinLinBU Feel free to submit this change in a new PR